commons/zenoh-shm/src/posix_shm/segment.rs

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use std::fmt::{Debug, Display};
+
+use rand::Rng;
+use shared_memory::{Shmem, ShmemConf, ShmemError};
+use zenoh_result::{bail, zerror, ZResult};
+
+const SEGMENT_DEDICATE_TRIES: usize = 100;
+
+/// Segment of shared memory identified by an ID
+pub struct Segment<ID> {
+    pub shmem: Shmem,
+    pub id: ID,
```

Comment:
Since `ID` is a const generic I believe there is no need to add `ID` on the stack.
One alternative is to have the `ID` as a const like done in the [protocol extensions](https://github.com/eclipse-zenoh/zenoh/blob/0943243da1a9eb14644897b13934be25610c4a94/commons/zenoh-protocol/src/common/extension.rs#L114).